bigchaindb/backend/mongodb/query.py

```diff
@@ -17,7 +17,10 @@
 @register_query(MongoDBConnection)
 def write_transaction(conn, signed_transaction):
     try:
-        return conn.db['backlog'].insert_one(signed_transaction)
+        return conn.run(
+            lambda db:
```

Comment:
The linter complains, so no.

We can have one of the following:
```python
return conn.run(lambda db:
                db['backlog'].insert_one(signed_transaction))
```

or 

```python
return conn.run(
    lambda db:
    db['backlog'].insert_one(signed_transaction))
```

I've chosen the latter because this:
```python
@register_query(MongoDBConnection)
def get_transaction_from_block(conn, transaction_id, block_id):
    try:
        return conn.run(
            lambda db:
            db['bigchain']
            .aggregate([
                {'$match': {'id': block_id}},
                {'$project': {
                    'block.transactions': {
                        '$filter': {
                            'input': '$block.transactions',
                            'as': 'transaction',
                            'cond': {
                                '$eq': ['$$transaction.id', transaction_id]
                            }
                        }
                    }
                }}]).next()['block']['transactions'].pop())
    except (StopIteration, IndexError):
        # StopIteration is raised if the block was not found
        # IndexError is returned if the block is found but no transactions
        # match
        return
```

looks better than this:
```python
@register_query(MongoDBConnection)
def get_transaction_from_block(conn, transaction_id, block_id):
    try:
        return conn.run(lambda db:
                        db['bigchain']
                        .aggregate([
                            {'$match': {'id': block_id}},
                            {'$project': {
                                'block.transactions': {
                                    '$filter': {
                                        'input': '$block.transactions',
                                        'as': 'transaction',
                                        'cond': {
                                            '$eq': ['$$transaction.id', transaction_id]
                                        }
                                    }
                                }
                            }}]).next()['block']['transactions'].pop())
    except (StopIteration, IndexError):
        # StopIteration is raised if the block was not found
        # IndexError is returned if the block is found but no transactions
        # match
        return
```

> Another idea: Use a decorator like @retry and then just return the lambda directly :)

not sure on how this can work.